README
Network Manager is an application that provides an overview of a wireless network from the point of view of the router. 
It is intended for use by home users, as its feature set is too limited for use in a business environment with a large
network.

vm/CHANGELOG
-------------------------------------------
Weathermap4rrd ** PHP and PERL ** CHANGELOG
-------------------------------------------

1.2b final
	PHP
		BUGS FIXED
			- fix a security issue notified by Alla Bezroutchko (if register_global is set to ON in php.ini it could be exploited to hack server)
			- fix config file path detection to avoid few errors
			- add RRDTool 1.3 support

1.2 final
	PHP
		BUGS fixed
			- text display of bandwidth usage in rrdtool graph was displayed in bytes instead of bits on second line
			- add weathermap.conf.example file
			- translate a few french error messages to english language
			- fix bug when no UNIT directive used (thank you to Ardeck)
			- fix bug : IPCHECK directive was not detected and had no effect
			- fix bug : file defined in HTMLFILE can't be create the first time
			- fix bug : REFRESH directive had no effect. 60s was hardcoded (thanks to wasu)
			
1.2 RC8

	PHP
		BUGS fixed
			DS name in RRDTool file were stored in code. Detect DS names from position in RRD file.
			hardcoded call to rrdtool in rrd.php fixed.
			histo.php called index.php instead of w4rrd.php since 1.2RC6 release
			
			
		Enhancement
			Graph are displayed in bits rather than in bytes
			add checks for debugging in graph generation
			add capability to display your own content in overlib window like with
						LINKOVERLIBGRAPH http://www.google.fr/intl/fr_fr/images/logo.gif
					or
						LINKOVERLIBGRAPH currentvalues.php (where currentvalues.php could be replaced by your own php script)
						You have to set up php.ini file with allow_url_fopen = On. For security reason this is often disabled. 
						See http://fr.php.net/manual/en/ref.filesystem.php#ini.allow-url-fopen for details.


			disable label display if no label is set in configuration file
			allow to choose display node with a circle instead of usual box (LABELTYPE round)
			allow to disable display of number on internodes rounds (INTERNODENUMDISP directive)
			allow to disable display of color round for middle internode (INTERNODEDISPLAYMID directive)
			allow to use different source files for input and output values of a link (TARGETIN and TARGETOUT directives)
			new legend display available (LEGENDSTYLE line)
			new type of link arrows available (ARROW halfarrow)


1.2 RC7

	PHP

		BUGS fixed
			height and width map display that created a blank screen with Internet Explorer
			config variable was not take into account by index.php. It was only with w4rrd.php.

			Thank you to Dan Lowry for his contribution to fix theses bugs

1.2 RC6 

	PHP
		BUGS fixed
			sometimes last value of RRDTool 1.2 database was not found

		NEW FEATURES
			add link bandwidth graph popup over map (LINKOVERLIBGRAPH,LINKINFOURL,HTMLFILE directives)
				(thank you to Bertrand Schvallinger and others that send me some ideas or code). 
				WARNING : you need to copy overlib_mini.js to your webserver docroot. This file is used by index.php.
				
			add capability of choosing configuration filename with "config" variable (example : http://myserver/weathermap/index.php?config=internet.conf )
				(thank you to hoffswell for his simple but useful modification)
			add node check (IP and CHECK directives)
				You are able to test node presence in network (ping or tcp port check available). 
				I suggest to use TCP check rather than ping because it is really faster if node is unreachable.
			add new legend display available (use LEGENDSTYLE directive to choose between "original" or "line" style)
			add LEGENDPOS directives to define LEGEND position in map (same as KEYPOS that is still available)
			add REFRESH directive to specify period between refresh of display (default is 60 seconds)
				

1.2 RC5 30th May, 2005

	PHP :
		BUGS fixed
		fix % value displayed if > 100%
		fix an important bug about reading values in RRD file 
		fix error when ICONRESIZE too low to create image bounds greater than 1 pixel
		fix error when no icon are defined in configuration file
		fix circle bad display in some case
		
		NEW FEATURES
		add rrdstep parameters in config.php script (this parameter is used for RRDTool 1.2. You should set ip up with the base interval in seconds with which data have been fed into your RRD file
		add GROUP directive support to PHP version
		add MRTG reading capability (if .htm or .html extension is detected in TARGET)
		add FORCEMRTG directive to force TARGET to be treated as a MRTG link
		add BACKGROUNDCOLOR directive to define a background color on map
		add capability to generate graph for past date (see Documentation for details)
		add histo.php to generate animation of past bandwidths usage (let me know if you find this option useful)

		OTHER
		few minors improvement
		improve error message display

		Special thanks to Mitio Matanov for his support and his suggestions.

V1.2 RC4
	PHP :
		fix black title background display when no TITLEFOREGROUND and TITLEBACKGROUND are not used
		fix bug with arrow "circle" mode. 
		fix bug with RRDTool v1.2
	
V1.2 RC3 21st May 2005
	Perl release :
		add RRDTool v1.2 support
		fix black title background display when no TITLEFOREGROUND and TITLEBACKGROUND are not used
		add INTERNODEDISPLAY directive to display number of internode on graph

	PHP :
		add RRDTool v1.2 support
		fix black title background display when no TITLEFOREGROUND and TITLEBACKGROUND are not used
		add INTERNODEDISPLAY directive to display number of internode on graph

V1.2 RC2b 6th May 2005
	Perl release :
		fix bug for IN value link display (thanks to hoffswell)

V1.2 RC2 6th May 2005
	Perl release :
		fix bug for some use of internodes (thanks to Alexandre Modesto)
		fix bug : normal arrow was not displayed correctly when no INTERNODE defined (thanks to Dan Lowry)
		fix bug : AUTOSCALE directive behavior was not correct compared to PHP version (thanks to hoffswell)

v1.2 RC1 4th May 2005

	fix bug for black background when no BACKGROUND directive is used.

vm/config.php
<?php

$rrdtool = "/usr/bin/rrdtool";
//$rrdtool = "/usr/src/rrdtool-1.3.6/src/rrdtool";

# You have to set up this variable to your weathermap4rrd path
# it will be used to read config file
$weathermap4php="/var/www/html/weathermap";
//$configfile=$weathermap4php."/weathermap.conf";
echo $configfile;
$configfile=$_GET['config'];
if (empty($configfile)) {
	if (empty($weathermap4php)) {
	    $configfile = "weathermap.conf";
	} else {
	    $configfile = $weathermap4php."/weathermap.conf";
	}
} else {
	if (empty($weathermap4php)) {
	    $configfile = $configfile;
	} else {
	    $configfile = $weathermap4php."/".$configfile;
	}
} 
#echo "configfile = $configfile";
	
$adminpage="index.php";
$VERSION="1.2b";
$DEBUG=0;

# Size of map if nothing defined in configuration file
$width=600;
$height=700;

# Default font used if none is defined in configuration file (FONT directive)
$font=4; 

# Default width base arrow if none is defined in configuration file (ARROW directive)
$width_arrow=4;

# Default colors of title background and forecolor if non are specified in configuration file (TITLEBACKGROUND and TITLEFOREGROUND directives)

# White background
$titlebackground_red=255;
$titlebackground_green=255;
$titlebackground_blue=255;

# Black foreground writing
$titleforeground_red=0;
$titleforeground_green=100;
$titleforeground_blue=0;

# Default background color if non is specified in configuration file (BACKGROUNDCOLOR directive)
$backgroundcolor_red=255;
$backgroundcolor_green=255;
$backgroundcolor_blue=255;

# Specifies the base interval in seconds with which data have been fed into the RRD file
# This value is only used for RRDTool 1.2 binaries
$rrdstep=300; # = 5 minutes

# Period between refresh of display (in seconds) if none is defined in weathermap.conf 
$refresh_display=60;

# Type of javascript library used of rollover above map elements
$ROLLOVERLIB="overlib";


?>
